src/Shouldly/Internals/EqualityComparer.cs

```diff
@@ -33,10 +33,10 @@ public bool Equals(T x, T y)
             // Null?
             if (!type.IsValueType() || (type.IsGenericType() && type.GetGenericTypeDefinition().IsAssignableFrom(NullableType)))
             {
-                if (object.Equals(x, default(T)))
-                    return object.Equals(y, default(T));
+                if (object.Equals(x, null))
+                    return object.Equals(y, null);
 
-                if (object.Equals(y, default(T)))
+                if (object.Equals(y, null))
```

Comment:
`default(T)` is always `null` given the `if` condition. This is nicer than doing `default(T)!`.